android/src/firebase/config/TitaniumFirebaseConfigModule.java

```diff
@@ -4,148 +4,177 @@
+        if (params instanceof String) {
+            FirebaseRemoteConfig.getInstance().setDefaultsAsync(TiRHelper.getResource("xml." + TiConvert.toString(params)));
+        } else if (params instanceof HashMap) {
+            FirebaseRemoteConfig.getInstance().setDefaultsAsync((HashMap) params);
+        } else {
+            throw new Exception("Invalid defaults provided. Please either pass a dictionary or string");
+        }
+    }
+
+    @Kroll.method
+    public void activateFetched(KrollFunction callback) {
+        FirebaseRemoteConfig.getInstance().activate().addOnCompleteListener(task -> {
+            if (callback != null) {
+                KrollDict result = new KrollDict();
+                if (task.isSuccessful()) {
+                    result.put(PROPERTY_SUCCESS, true);
+                    result.put(PROPERTY_RESULT, task.getResult());
+                } else {
+                    result.put(PROPERTY_SUCCESS, false);
+                    if (task.getException() != null) {
+                        result.put(PROPERTY_ERROR, task.getException().getMessage());
+                    }
+                }
+                callback.callAsync(getKrollObject(), result);
+            }
+        });
+    }
+
+    @Kroll.method
+    public KrollDict configValueForKey(String key) {
+        String stringValue = FirebaseRemoteConfig.getInstance().getString(key);
+        boolean boolValue = FirebaseRemoteConfig.getInstance().getBoolean(key);
+        double numberValue = FirebaseRemoteConfig.getInstance().getDouble(key);
+
+        KrollDict result = new KrollDict();
+        result.put("bool", boolValue);
+        result.put("number", numberValue);
+
+        if (stringValue != null) {
+            result.put("string", stringValue);
+        }
+
+        return result;
+    }
+
+    @Kroll.setProperty
+    public void enableRealtimeUpdates(Boolean value) {
+        if (value) {
+            if (configUpdateListener == null) {
+                configUpdateListener = FirebaseRemoteConfig.getInstance().addOnConfigUpdateListener(new ConfigUpdateListener() {
+                    @Override
+                    public void onUpdate(ConfigUpdate configUpdate) {
+                        Set<String> keys = configUpdate.getUpdatedKeys();
+                        KrollDict result = new KrollDict();
+                        result.put("keys", keys.toArray());
+                        fireEvent("update", result);
+
+                        FirebaseRemoteConfig.getInstance().activate().addOnCompleteListener((OnCompleteListener) task -> {
+                        });
+                    }
+
+                    @Override
+                    public void onError(FirebaseRemoteConfigException error) {
+                        Log.e("FirebaseConfig", error.getMessage());
+                    }
+                });
+            }
+        } else if (configUpdateListener != null) {
+            configUpdateListener.remove();
```

Comment:
It should only go to the the else case if you call it with `FirebaseConfig.enableRealtimeUpdates = false;` and then if there is a listener it will remove it, if not it won't do anything.

I tried it with
```
if (configUpdateListener != null) {
    configUpdateListener.remove();
}
```
at first and it still added 4 listeners when I call
```
	FirebaseConfig.enableRealtimeUpdates = true;
	FirebaseConfig.enableRealtimeUpdates = true;
	FirebaseConfig.enableRealtimeUpdates = true;
	FirebaseConfig.enableRealtimeUpdates = true;
```

but I'll check it again in a few minutes.

Reply:
Interesting. Maybe it has to be nulled as well on Android. I mean, if this approach works, it's fine! :-)

Reply:
Think we keep it the current way :smile: 

Even using that before the register:
```
if (configUpdateListener != null) {
   configUpdateListener.remove();
   configUpdateListener = null;
}
```
 will still add 4 listeners. Strange, maybe a timing issue? :shrug: The current version works better. I can also remove and readd it fine